src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -126,86 +106,60 @@ private TextFormattingRunProperties Format
             }
         }
 
-        private void InvalidateCache()
-        {
-            _taggerProvider.ThreadingContext.ThrowIfNotOnUIThread();
-            _cacheSnapshot = null;
-            _cache.Clear();
-        }
-
-        IEnumerable<ITagSpan<IntraTextAdornmentTag>> ITagger<IntraTextAdornmentTag>.GetTags(NormalizedSnapshotSpanCollection spans)
-            => GetTags(spans);
-
-        public IReadOnlyList<TagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        public override void AddTags(
+            NormalizedSnapshotSpanCollection spans,
+            SegmentedList<TagSpan<IntraTextAdornmentTag>> adornmentTagSpans)
         {
             try
             {
                 if (spans.Count == 0)
-                    return [];
+                    return;
 
+                // If the snapshot has changed, we can't use any of the cached data, as it is associated with the
+                // original snapshot they were created against.
                 var snapshot = spans[0].Snapshot;
-                if (snapshot != _cacheSnapshot)
-                {
-                    // Calculate UI elements
-                    _cache.Clear();
```

Comment:
this logic was busted.  we only cleared the cache when the snapshot changed.  but this is a view tagger.  the snapshot may not change while the view *does* change.   but we wouldnt' know to then get teh new tags.